### PR TITLE
Refactor DXDVotingMachine

### PR DESCRIPTION
**Pull request by AugustoL**
This PR is big and it involves changes in the DXDVM and also in the Scheme contract, the main changes are fixes in how the staking token and gas refund balance is managed by each avatar and scheme, handling it everything with schemes ids (previously org ids) the scheme id is `keccak256(avatar, scheme)`.
It also changed how the states are used in the DXDVM and Scheme, now the DXDVM will execute the proposal only once, and it will call the finishProposal function after that. This allows the DXDVM to execute a proposal and if it reverts it can catch the exception and log it. This is why we have to finish the proposal after that, to change the state in storage without risks of revert and loosing storage changes in the scheme.
Also the maxSecondsFor execution was removed since the proposal will execute only once and if it fails it will be finalized anyways, so no need to have that extra hacky parameter.
In addition I removed configuration parameters that we never used and unnecessary functions and variables in DXDVM.

- Remove unused voting params in dxd voting machine
  - Remove voteOnBehalf, voterReputationLossRatio and activationTime in voting parameters
- Remove unnecessary mint/burn in DXDVotingMachineCallbacks
- Remove unnecessary voter param in vote function in the dxdvotingmachine
- Change comments and order of NO and YES options in dxdVotingMachine
- Remove claimStakingTokens and proposeMultipleChoice functions in dxdVM
- Refactor organizations data structure in DXDVM
  - Move avatar address, averageDownstakesOfBoosted and orgBoostedCnt to organization structure and keep
track of staked token balance for each organization, not allowing to create fake orgs to steal
staked tokens in other organizations.
- Remove avatarOwner and GEN staking contracts used by default
- Use base 1000 for queuedVoteRequiredPercentage
- Move boostedVoteRequiredPercentage to voting params
- Try/catch proposal execution to avoid stuck failed proposals
- Divide staking token and vote gas balance per dao and scheme
- Remove executeBoosted function
- Refactor(contact/dao): change organization for scheme in names, each dao has multiple schemes
- Remove unnecessary maXSecondsForExecution in scheme and execution states
  - Since we are catching the revert of proposal execution in the DXVM the maxSecondsForExecution in
Scheme became obsolete, the scheme now has state Passed as final and the execution state can be
obtained from DXVM where we added Failed as ExecutionState and ExecutedInQueue and ExecutedInBoost
as proposal states
- Use only EIP 712 hash for signed operations in DXDVM
- Finish proposal in scheme form VM after proposal execution
  - This change allows the executeProposal function to fail anywhere and catch the error message,
before this if the proposal execution threw an error we lost thw changes done in storage, so teh
the state was left as submitted, so first, we execute the proposal and then we finish, the finishProposal
function wont throw and it sets the proposal in a finalized state
- Change WalletScheme for Scheme in error messages in Scheme
- Return error message in string instead of bytes in ProposalExecuteResult
- Fix(contracts/dao): fix the wrong condition that returns proposal exec success in execute func
- Change AvatarScheme executeProposal to follow Scheme executeProposal changes
- Change ExpiredInQueue to Expired in DXVM
- Use EIP712 for action signature verification

### CODE REVIEW

File: contracts/dao/schemes/Scheme.sol [hunk 1]
```diff
@@ -291,11 +255,34 @@ abstract contract Scheme is DXDVotingMachineCallbacks {
...
+     * @return bool success
+     */
+    function finishProposal(bytes32 _proposalId, uint256 _winningOption)
+        public
+        virtual
+        onlyVotingMachine
+        returns (bool)
+    {
+        Proposal storage proposal = proposals[_proposalId];
+        require(proposal.state == ProposalState.Submitted, "Scheme: must be a submitted proposal");
```
Comment (MiltonTulli) [inline contracts/dao/schemes/Scheme.sol:276]:
```suggestion       
        if (proposal.state != Proposal.Submitted) {
           revert Scheme__ProposalMustBeSubmitted();  // create new error definition
        }

```

File: contracts/dao/votingMachine/DXDVotingMachine.sol [hunk 2]
```diff
@@ -345,24 +287,24 @@ contract DXDVotingMachine {
...
      *    _params[5] -_quietEndingPeriod
      *    _params[6] -_proposingRepReward
-     *    _params[7] -_votersReputationLossRatio
-     *    _params[8] -_minimumDaoBounty
-     *    _params[9] -_daoBountyConst
-     *    _params[10] -_activationTime
-     * @param _voteOnBehalf - authorized to vote on behalf of others.
+     *    _params[7] -_minimumDaoBounty
+     *    _params[8] -_daoBountyConst
+     *    _params[9] - _boostedVoteRequiredPercentage
```
Comment (MiltonTulli) [inline contracts/dao/votingMachine/DXDVotingMachine.sol:292]: NatSpec comments?

File: contracts/dao/votingMachine/DXDVotingMachine.sol [hunk 3]
```diff
@@ -587,127 +507,88 @@ contract DXDVotingMachine {
...
-                .toEthSignedMessageHash();
-        }
-        address staker = delegationDigest.recover(_signature);
-        //a garbage staker address due to wrong signature will revert due to lack of approval and funds.
-        require(staker != address(0), "staker address cannot be 0");
-        require(stakesNonce[staker] == _nonce);
+        bytes32 stakeHashed = hashAction(proposalId, staker, stakeDecision, amount, nonce, 2);
+        address staker = stakeHashed.recover(signature);
+        require(staker == stakeHashed.toEthSignedMessageHash().recover(signature), "wrong signer");
+        require(stakesNonce[staker] == nonce);
```
Comment (MiltonTulli) [inline contracts/dao/votingMachine/DXDVotingMachine.sol:531]: Can you add error msg here?

**Review by MiltonTulli — APPROVED** [review]
Looks good to me. I just left some format comments but we can address them in a separate pr.

File: contracts/dao/votingMachine/DXDVotingMachine.sol (hunk 3 above)
Comment (AugustoL) [reply contracts/dao/votingMachine/DXDVotingMachine.sol:531]: We do it in next PR

**Pull request merged**